scalyr_agent/builtin_monitors/kubernetes_events_monitor.py

```diff
@@ -0,0 +1,478 @@
+        """
+
+        oldest_time = datetime.datetime.utcnow()
+        oldest = None
+
+        # loop over all nodes and find the one with the oldest create time
+        for node in nodes:
+            metadata = node.get( 'metadata', {} )
+            create_time = metadata.get( 'creationTimestamp', None )
+            if create_time is None:
+                continue
+
+            name = metadata.get( 'name', None )
+            if name is None:
+                continue
+
+            # skip the master node if necessary
+            labels = metadata.get( 'labels', {} )
+            if ignore_master and 'node-role.kubernetes.io/master' in labels:
+                continue
+
+            #convert to datetime
+            create_time = scalyr_util.rfc3339_to_datetime( create_time )
+
+            # if we are older than the previous oldest datetime, then update the oldest time
+            if create_time is not None and create_time < oldest_time:
+                oldest_time = create_time
+                oldest = name
+
+        return oldest
+
+    def _check_nodes_for_leader( self, k8s, query_fields='' ):
+        """
+            Checks all nodes on the cluster to see which one has the oldest creationTime
+            If `ignore_master` is True then any node with the label `node-role.kubernetes.io/master`
+            is ignored.
+        """
+        response = k8s.query_api( '/api/v1/nodes%s' % query_fields )
+        nodes = response.get( 'items', [] )
+
+        # sort the list by node name to ensure consistent processing order
+        # if two nodes have exactly the same creation time
+        nodes.sort( key=lambda n: n.get( 'metadata', {} ).get( 'name', '' ) )
+
+        # the leader is the longest running node
+        return self._get_oldest_node( nodes, self._ignore_master )
+
+    def _get_current_leader( self, k8s ):
+        """
+            Queries the kubernetes api to see which node is the current leader node.
+
+            If there is currently a leader node, then only that node is polled to see if it still
+            exists.
+
+            Check to see if the leader is specified as a node label.
+            If not, then if there is not a current leader node, or if the current leader node no longer exists
+            then query all nodes for the leader
+        """
+
+        try:
+            leader = None
+            # first check to see if the leader node is specified as a label
+            # we do this first in case the label has been moved to a different node but the old node is still
+            # alive
+            if self._check_labels:
+                query = "?labelSelector=%s" % (urllib.quote( 'agent.config.scalyr.com/events_leader=true' ) )
+                leader = self._check_nodes_for_leader( k8s, query )
+
+            # if no labelled leader was found, then check to see if the previous leader is still alive
+            if leader is None:
```

Comment:
Make this condition `if leader is not None` so that you can move the `leader set from label` up here.

Reply:
Done.